src/main/java/at/hannibal2/skyhanni/features/misc/customscoreboard/ScoreboardEvents.kt

```diff
@@ -0,0 +1,499 @@
+            ?.startsWith(" Ends In:") == true) {
+        add(
+            "§6Mining Fiesta: " + TabListData.getTabList().nextAfter("§e§lEvent: §r§6Mining Fiesta")
+                ?.removePrefix(" Ends In: ")
+        )
+    }
+
+    // Wind
+    getSbLines().firstOrNull { SbPattern.windCompassPattern.matches(it) }?.let {
+        add(it)
+        add("| ${getSbLines().first { SbPattern.windCompassArrowPattern.matches(it) }} §f|")
+    }
+
+    // Better Together
+    getSbLines().firstOrNull { SbPattern.nearbyPlayersPattern.matches(it) }?.let {
+        add("§dBetter Together")
+        add(" $it")
+    }
+
+    // Zone Events
+    if (getSbLines().any { SbPattern.miningEventPattern.matches(it) }
+        && getSbLines().any { SbPattern.miningEventZonePattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.miningEventPattern.matches(it) }.removePrefix("Event: "))
+        add("in ${getSbLines().first { SbPattern.miningEventZonePattern.matches(it) }.removePrefix("Zone: ")}")
+    }
+
+    // Zone Events but no Zone Line
+    if (getSbLines().any { SbPattern.miningEventPattern.matches(it) }
+        && getSbLines().none { SbPattern.miningEventZonePattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.miningEventPattern.matches(it) }
+            .removePrefix("Event: "))
+    }
+
+    // Mithril Gourmand
+    if (getSbLines().any { SbPattern.mithrilRemainingPattern.matches(it) }
+        && getSbLines().any { SbPattern.mithrilYourMithrilPattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.mithrilRemainingPattern.matches(it) })
+        add(getSbLines().first { SbPattern.mithrilYourMithrilPattern.matches(it) })
+    }
+
+    // Raffle
+    if (getSbLines().any { SbPattern.raffleTicketsPattern.matches(it) }
+        && getSbLines().any { SbPattern.rafflePoolPattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.raffleTicketsPattern.matches(it) })
+        add(getSbLines().first { SbPattern.rafflePoolPattern.matches(it) })
+    }
+
+    // Raid
+    if (getSbLines().any { SbPattern.yourGoblinKillsPattern.matches(it) }
+        && getSbLines().any { SbPattern.remainingGoblinPattern.matches(it) }) {
+        add(getSbLines().first { SbPattern.yourGoblinKillsPattern.matches(it) })
+        add(getSbLines().first { SbPattern.remainingGoblinPattern.matches(it) })
+    }
+}
+
+private fun getMiningEventsShowWhen(): Boolean {
+    return IslandType.DWARVEN_MINES.isInIsland() || IslandType.CRYSTAL_HOLLOWS.isInIsland()
+}
+
+private fun getDamageLines(): List<String> {
+    return listOf(getSbLines().first { SbPattern.bossHPPattern.matches(it) }) +
+        (getSbLines().first { SbPattern.bossDamagePattern.matches(it) })
+}
+
+private fun getDamageShowWhen(): Boolean {
+    return getSbLines().any { SbPattern.bossHPPattern.matches(it) || SbPattern.bossDamagePattern.matches(it) }
+}
+
+private fun getMagmaBossLines() = listOf(
```

Comment:
I dont get what you mean

The magma boss has different waves with multiple different lines

Reply:
yeah but since the patterns aren't used anywhere else the can be concatenated, before compiling, '|' operator

Reply:
hm I see what you mean
idk if one pattern for like 7 completely different lines is the right approach tho

Reply:
Your call. But if you want to keep these patterns, you need to turn the search around.
Currently you search the SBlines n times (n begin the amount of patterns).
But it could be once matching every line once against all patterns at the same time (early returning if the pattern has already matched).